sqlmesh/core/dialect.py

```diff
@@ -761,7 +761,7 @@ def text_diff(
 
 
 DIALECT_PATTERN = re.compile(
-    r"(model|audit).*?\(.*?dialect[^a-z,]+([a-z]*|,)", re.IGNORECASE | re.DOTALL
+    r"(model|audit).*?\(.*?dialect\s+'?([a-z]*)'?", re.IGNORECASE | re.DOTALL
```

Comment:
does this have performance implications?

Reply:
Maybe, but I'm not sure how else we can express this. Any ideas?